parsl/executors/high_throughput/mpi_executor.py

```diff
@@ -0,0 +1,105 @@
+"""A simplified interface for HTEx when running in MPI mode"""
+from typing import Optional, Tuple, List, Union, Callable, Dict
+from inspect import signature
+import re
+
+import typeguard
+
+from parsl.data_provider.staging import Staging
+from parsl.executors.high_throughput.executor import HighThroughputExecutor
+from parsl.executors.status_handling import BlockProviderExecutor
+from parsl.jobs.states import JobStatus
+from parsl.providers import LocalProvider
+from parsl.providers.base import ExecutionProvider
+
+
+class MPIExecutor(HighThroughputExecutor):
+    """A version of :class:`~parsl.HighThroughputExecutor` tuned for executing MPI tasks
+
+    Parameters
+    ----------
+    max_tasks: int
+        Maximum number of MPI applications to run at once per block
+    """
+
+    @typeguard.typechecked
+    def __init__(self,
+                 label: str = 'MPIExecutor',
+                 provider: ExecutionProvider = LocalProvider(),
+                 launch_cmd: Optional[str] = None,
+                 address: Optional[str] = None,
+                 worker_ports: Optional[Tuple[int, int]] = None,
+                 worker_port_range: Optional[Tuple[int, int]] = (54000, 55000),
+                 interchange_port_range: Optional[Tuple[int, int]] = (55000, 56000),
+                 storage_access: Optional[List[Staging]] = None,
+                 working_dir: Optional[str] = None,
+                 worker_debug: bool = False,
+                 max_tasks: int = 1,
+                 prefetch_capacity: int = 0,
+                 heartbeat_threshold: int = 120,
+                 heartbeat_period: int = 30,
```

Comment:
Do we need to enumerate all of base class' (`HighThroughputExecutor`) keyword arguments if we aren't modifying the defaults? My concern is if we update the base class, then we might forget to update `MPIExecutor`. It seems like a safer approach might be to only explicitly have new and updated keyword arguments and then pass any remaining `**kwargs` to the base class. Thoughts?

Reply:
Using **kwargs was my first thought too, but I wanted to try enumerating them to give better IDE support.

I made a unit test that will fail of the signature of the base class changes. Does that seem sufficient?

Reply:
Sounds reasonable to me!

(I really wish IDEs supported better inspection of classes...)